src/videojs-contrib-hls.js

```diff
@@ -327,49 +350,56 @@ export default class HlsHandler extends Component {
       this.tech_.audioTracks().addEventListener('change', this.audioTrackChange_);
     });
 
-    this.masterPlaylistController_.on('selectedinitialmedia', () => {
-      let audioTrackList = this.tech_.audioTracks();
-      let media = this.masterPlaylistController_.media();
-      let master = this.masterPlaylistController_.master();
-      let mediaGroups = master.mediaGroups;
-      let attributes = {
-        audio: {main: {default: true}}
-      };
-
-      if (!media.attributes) {
-        // source URL was playlist manifest, not master
-        // no audio tracks to add
+    this.masterPlaylistController_.on('audioinfo', (e) => {
```

Comment:
Originally it was at a lower level but it ended up with HLS trying to revert data on MediaSources which was a big jump. So it was determined that we should just fire 'audioinfo' every time we get it, and have the objects that care about the event act on it. In this case HLS.